server/repository/src/migrations/README.md

```diff
@@ -0,0 +1,47 @@
+
+Diesel dsl can be used in data migrations, however, for some operations sql statement are prefered, see `Raw SQL vs Diesel` below.
+
+## How to add migration
+
+Identify next version, see [package.json](../../../../package.json) for current version, then increment `patch` by one (we use semantic versioning syntax for our version number, but our app versioning wouldn't necesseraly follow SemVer guidelines which are aimed at publicly consumed packages and libraries, see [version.rs](./version.rs) for more details).
+
+**note** everything after `patch` is condiered `pre-release`, and cannot be upgraded further (pre-relase version is really undefined, it could be a test branch or a release candidate), but `pre-release` versions can be used to manually test migrations/functionality in production.
+
+Increment [package.json](../../../../package.json) version to the new version and create new migration folder with the version number. Copy template or existing migration, and rename to new version approprately. 
+
+Add new version mod to [root migrations mode](mod.rs) and add new version to `vec!` of visitors. Add actual migration code and tests, through tests you should be able to check sql syntax and data migration logic without starting server.
+
+## Raw SQL vs Diesel
+
+Ideally we would be using existing repositories for data migrations, but this will break as soon as new migrations change schema and repositories are updated. We can achieve almost any type of migration with raw sql statement, both data and schema, but some raw sql logic is hard to either read/write or keep consistent between sqlite and postgres. On the other hard, using diesel definitions comes with a lot of boilerplate, and can also be hard to read/write but it can help with serialisation and difference in sqlite and postgres syntax. Although at the time of writing this README.md there is no defined standard or guideline of when to use raw sql vs diesel dsl, some effort was made to show when one is better then the other in `/templates`. A quick summary:
+* In most cases, for schema migrations its easier to use raw sql statements, there is small variation in syntax, mainly types, but with a help of simple `sql!` macro and common [types](types.rs) it looks clean and trivial, see [adding table](templates/adding_table/mod.rs)
+* Inserting mock data sql looks the same, and we can use use diesel `.bind` to serialise more complex types like `NaiveDateTime`, as demonstrated in [data migration template](templates/data_migrations/mod.rs)
+* Since raw sql query still require a struct to save a result in, I found that adding diesel `table!` with minimum fields has about the same amount of code and allows for a way to select a result into a tuple and opens a way to use diesel dsl for queries and updates, [again in data migration template, query/update in migration and query in test](templates/data_migrations/mod.rs)
```

Comment:
:nit is this right?
```suggestion
* Since raw sql query still requires a struct to save a result in, I found that adding diesel `table!` with minimum fields has about the same amount of code and allows for a way to select a result into a tuple and opens a way to use diesel dsl for queries and updates, [again in data migration template, query/update in migration and query in test](templates/data_migrations/mod.rs)
```

Reply:
🙏